pennylane/ops/op_math/change_op_basis.py

```diff
@@ -0,0 +1,170 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines a class for compute-uncompute patterns.
+"""
+from collections import Counter
+from functools import reduce
+
+from pennylane import math, queuing
+from pennylane.decomposition import add_decomps, register_resources, resource_rep
+from pennylane.operation import (
+    DiagGatesUndefinedError,
+    EigvalsUndefinedError,
+    MatrixUndefinedError,
+    Operator,
+    SparseMatrixUndefinedError,
+)
+from pennylane.ops.op_math import adjoint
+
+from .composite import CompositeOp, handle_recursion_error
+
+
+def change_op_basis(compute_op: Operator, target_op: Operator, uncompute_op: Operator = None):
+    """Construct an operator that represents the product of the
+    operators provided.
+
+    Args:
+        compute_op (:class:`~.operation.Operator`): A single operator or product that applies quantum operations.
+        target_op (:class:`~.operation.Operator`): A single operator or a product that applies quantum operations.
+        uncompute_op (None | :class:`~.operation.Operator`): An optional single operator or a product that applies quantum
+            operations. ``None`` corresponds to ``uncompute_op=qml.adjoint(compute_op)``.
+
+    Returns:
+        ~ops.op_math.ChangeOpBasis: the operator representing the compute-uncompute pattern.
+
+    .. seealso:: :class:`~.ops.op_math.ChangeOpBasis`
+    """
+
+    return ChangeOpBasis(compute_op, target_op, uncompute_op)
+
+
+class ChangeOpBasis(CompositeOp):
+    """
+    Composite operator representing a compute-uncompute pattern of operators, which constitutes changing the basis in
+    which an operator is applied.
+
+    Args:
+        compute_op (:class:`~.operation.Operator`): A single operator or product that applies quantum operations.
+        target_op (:class:`~.operation.Operator`): A single operator or a product that applies quantum operations.
+        uncompute_op (:class:`~.operation.Operator`): A single operator or a product that applies quantum operations.
+            Default is uncompute_op=qml.adjoint(compute_op).
+
+    Returns:
+        (Operator): Returns an Operator which is the change_op_basis of the provided Operators: compute_op, target_op, uncompute_op.
+
+    .. seealso:: :func:`~.change_op_basis`
+    """
+
+    def __init__(self, compute_op: Operator, target_op: Operator, uncompute_op: Operator = None):
+        if uncompute_op is None:
+            uncompute_op = adjoint(compute_op)
+        super().__init__(compute_op, target_op, uncompute_op)
```

Comment:
```suggestion
        super().__init__(uncompute_op, target_op, compute_op)
```
So that the string representation is in the correct order. Right now we have
```python
def circuit():
    qml.change_op_basis(qml.X(0), qml.H(0))
```
```pycon
>>> print(qml.draw(circuit)())
0: ──X@H@(X†)─┤  
```
which looks like the uncompute op is applied first.

Reply:
Aha ok thanks.